homePage.js

```diff
@@ -0,0 +1,273 @@
+const DOM = {
+    content: null,
+    popUp: null,
+    mainHtmlBody: null,
+    closePopUp: null,
+    searchInput: null,
+    searchContainer: null
+}
+
+let coinsData = [];
+let setIntervalId;
+const cachedCoins = [];
+const selectedCoins = [];
+
+const Crypto_API = `https://api.coingecko.com/api/v3/coins`
+
+class CryptoCoinMaker {
+    constructor(id, symbol, name) {
+        this.id = id
+        this.symbol = symbol
+        this.name = name
+    }
+}
+
+DOM.mainHtmlBody = document.querySelector("body")
+
+function showOrHideSearchContainer(status) {
+    DOM.searchContainer = document.querySelector("#searchContainer")
+    DOM.searchContainer.style.visibility = status
+}
+
+
+async function getCryptoCoinsInit() {
+
+    clearInterval(setIntervalId);
+
+    DOM.mainHtmlBody.style.backgroundImage = "url('./assests/bitcoin-sans-internet.jpg')";
+    DOM.content = document.querySelector("#coinsCardsContent")
+    DOM.popUp = document.querySelector("#staticBackdrop")
+    DOM.closePopUp = document.querySelector("#closePopUp")
+    showOrHideSearchContainer("visible")
+    DOM.searchInput = document.querySelector("#searchInput")
+    drawLoader()
+
+    try {
+        const url = `${Crypto_API}/list`
+        const result = await fetch(url);
+        const jsonResult = await result.json();
+        const someHundread = jsonResult.slice(10000, 10100);
+        const modeledResult = cryptoCoinMakerFn(someHundread)
+        coinsData = modeledResult
+        draw(coinsData)
+
+    } catch (ex) {
+        console.log(ex)
+        const swalTitleText = "Something Went Wrong :(";
+        const swalBodyText = "Try to Refresh the Page";
+        const swalIconType = "error"
+        swal(swalTitleText, swalBodyText, swalIconType, {
+            closeOnClickOutside: false,
+        })
+    }
+}
+
+getCryptoCoinsInit()
+
+function handleSearch() {
+    const inputValue = DOM.searchInput.value
+    if (!inputValue) {
+        getCryptoCoinsInit()
+    } else {
+        const filteredCoins = coinsData.filter((coin) => coin.symbol.toLowerCase() === inputValue.toLowerCase())
```

Comment:
inputValue.toLowerCase() can be one line above

and then the loop (filter_) will not need to execute the toLower on every iteration